backend/src/services/payments/payments-advisory-lock.ts

```
@@ -0,0 +1,41 @@
+  } finally {
+    try {
+      await client.query(UNLOCK_SQL_BY_MODE[mode], [lockName]);
+      client.release();
+    } catch (unlockError) {
+      logger.error('Failed to release payments advisory lock', {
+        lockName,
+        mode,
+        error: unlockError instanceof Error ? unlockError.message : String(unlockError),
+      });
+      client.release(true);
+      throw unlockError;
```

Biome (2.4.13)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)